woocommerce/woocommerce-ios
# [Mobile Payments] Check reader connection in Card Present Payments readiness

Part of: 
- #6607 


### Description


On opening Order Details, we checks the onboarding state for In-Person-Payments, and show onboarding if required.

This PR builds on #6761 to prevent the onboarding check if there is already a card reader connected. 
- #6761 

#### Note on testing

This commit removes the check for a card-present eligible payment gateway account which used to be part of the decision whether to show the `Collect Payment` button on Order Details or not. I did this to make the changes easier to test, and to be closer in line with Android and expected direction of the changes to the `Collect Payment` button.

To restore the previous behaviour, we will revert 318440cd before merging.


### Testing instructions


Using a store with both Stripe and WCPay plugins installed and activated

1. On the orders tab, tap + and use Create Order
2. Add a product, and tap next to go through to Order Details
3. Tap `Collect Payment`
4. Observe that you see the "Multiple Conflicting Plugins" onboarding screen
5. Use the `Manage Plugins` CTA and disable one of the payment plugins
6. Tap `Done`
7. Observe that you see an onboarding loading screen, which then dismisses and returns to the Order Details
8. Observe that the Collect Payment flow starts.
9. Connect a reader and collect payment.
10. On another device, activate the plugin you disabled again.
11. Repeat the above steps
12. Observe that onboarding is **not** checked or shown, and the collect payment screen is shown anyway

N.B. This is not an expected user flow, it's just to demonstrate that onboarding wasn't checked the second time around.

This can also be checked with a breakpoint in [CardPresentPaymentsOnboardingUseCase](https://github.com/woocommerce/woocommerce-ios/blob/73b72dd8b2a865199a00ae7c856c08f316d9b5ea/WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person%20Payments/CardPresentPaymentsOnboardingUseCase.swift#L59)

Testing that onboarding is shown correctly from Settings > In-Person Payments is also worthwhile here.

### Screenshots



https://user-images.githubusercontent.com/2472348/166473920-a14ced6e-e587-4b42-8d5a-d58f75d72277.mp4


---
